strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java

```diff
@@ -200,14 +202,173 @@ public void download(@ApiParam(value = "The storageId", required = true)
+    @RequestMapping(value = { "{storageId}/{repositoryId}/{path:.+}" }, method = RequestMethod.HEAD)
+    public void getHeaders(@ApiParam(value = "The storageId", required = true)
+                         @PathVariable String storageId,
+                         @ApiParam(value = "The repositoryId", required = true)
+                         @PathVariable String repositoryId,
+                         @RequestHeader HttpHeaders httpHeaders,
+                         @PathVariable String path,
+                         HttpServletRequest request,
+                         HttpServletResponse response)
+            throws Exception
+    {
+        logger.debug("Requested Headers for /" + storageId + "/" + repositoryId + "/" + path + ".");
+
+        Storage storage = configurationManager.getConfiguration().getStorage(storageId);
+        if (storage == null)
+        {
+            logger.error("Unable to find storage by ID " + storageId);
+
+            response.sendError(INTERNAL_SERVER_ERROR.value(), "Unable to find storage by ID " + storageId);
+
+            return;
+        }
+
+        Repository repository = storage.getRepository(repositoryId);
+        if (repository == null)
+        {
+            logger.error("Unable to find repository by ID " + repositoryId + " for storage " + storageId);
+
+            response.sendError(INTERNAL_SERVER_ERROR.value(),
+                               "Unable to find repository by ID " + repositoryId + " for storage " + storageId);
+            return;
+        }
+
+        if (!repository.isInService())
+        {
+            logger.error("Repository is not in service...");
+
+            response.setStatus(HttpStatus.SERVICE_UNAVAILABLE.value());
+
+            return;
+        }
+
+        if (repository.allowsDirectoryBrowsing() && probeForDirectoryListing(repository, path))
+        {
+            try
+            {
+                generateDirectoryListing(repository, path, request, response);
+            }
+            catch (Exception e)
+            {
+                logger.debug("Unable to generate directory listing for " +
+                             "/" + storageId + "/" + repositoryId + "/" + path, e);
+
+                response.setStatus(INTERNAL_SERVER_ERROR.value());
+            }
+
+            return;
+        }
+        
+        ArtifactInputStream is;
+        try
+        {
+            is = (ArtifactInputStream) getArtifactManagementService().resolve(storageId, repositoryId, path);
+            if (is == null)
+            {
+                response.setStatus(NOT_FOUND.value());
+                return;
+            }
+
+            if (isRangedRequest(httpHeaders))
+            {
+                logger.debug("Detecting range request....");
+
+                handlePartialDownload(is, httpHeaders, response);
+            }
+
+            artifactEventListenerRegistry.dispatchArtifactDownloadingEvent(storage.getId(), repository.getId(), path);
```

Comment:
The flow here with the dispatching of these two events doesn't look right.

Reply:
I do not get what you mean by 'flow'.
It's the same steps as in GET only removed the copying of artifact to Response Body

Reply:
Oh, so the code for this download has now been extracted inside a listener?